### ISSUE TITLE
Question regarding load importance loss calculation

### ISSUE THREAD

**Issue by wangyirui** [jinga-lala/DAMEX#3]
Thanks for the excellent work and releasing the code. A question raised when studying your code for calculating the load importance loss.

In your [code](https://github.com/jinga-lala/DAMEX/blob/main/models/dino/moe.py#L281), the first two arguments passed to Microsoft Tutel loss implementation is `F.softmax(logits[fg_mask, :], dim=1)` and `logits.gather(index=topk_ids, dim=1)[fg_mask, :] `. In [Tutel](https://github.com/microsoft/tutel/blob/main/tutel/impls/losses.py#L22), these two arguments are used to calculate the diff via `diff = scores_wo_noise.float() - threshold.float()`. My question is: why you use the prob scores _after softmax_ and the logits _without softmax_ to calculate the diff? 

Looking forward to your reply! Thanks!

**Comment by jinga-lala** [comment]
That's an interesting question @wangyirui. However, my changes for MoE loss only involves adding a `fg_mask` in the calculation for better gradient signals in Object Detection task (more details in paper). The answer to your question can be answered by someone from Tutel team. [Here ](https://github.com/microsoft/tutel/blob/125e6c601861b965ea9db039afff68ba04a887b3/tutel/impls/moe_layer.py#L281)is a link to their implementation which also takes in prob scores after softmax as input.
Keep me posted what you hear from them!
